controller/monitor/environment_check_monitor.go

```diff
@@ -203,66 +201,49 @@ func (m *EnvironmentCheckMonitor) syncPackagesInstalled(kubeNode *corev1.Node, n
 	case strings.Contains(osImage, "rocky"):
 		fallthrough
 	case strings.Contains(osImage, "ol"):
-		queryPackagesCmd = "rpm"
-		options = append(options, "-q")
-		packages = append(packages, "nfs-utils", "iscsi-initiator-utils", "cryptsetup", "device-mapper")
+		packageProbeExecutables["nfs-utils"] = "mount.nfs"
+		packageProbeExecutables["iscsi-initiator-utils"] = "iscsiadm"
+		packageProbeExecutables["cryptsetup"] = "cryptsetup"
+		packageProbeExecutables["device-mapper"] = "dmsetup"
 	case strings.Contains(osImage, "suse"):
-		queryPackagesCmd = "rpm"
-		options = append(options, "-q")
-		packages = append(packages, "nfs-client", "open-iscsi", "cryptsetup", "device-mapper")
+		packageProbeExecutables["nfs-client"] = "mount.nfs"
+		packageProbeExecutables["open-iscsi"] = "iscsiadm"
+		packageProbeExecutables["cryptsetup"] = "cryptsetup"
+		packageProbeExecutables["device-mapper"] = "dmsetup"
 	case strings.Contains(osImage, "arch"):
-		queryPackagesCmd = "pacman"
-		options = append(options, "-Q")
-		packages = append(packages, "nfs-utils", "open-iscsi", "cryptsetup", "device-mapper")
+		packageProbeExecutables["nfs-utils"] = "mount.nfs"
+		packageProbeExecutables["open-iscsi"] = "iscsiadm"
+		packageProbeExecutables["cryptsetup"] = "cryptsetup"
+		packageProbeExecutables["device-mapper"] = "dmsetup"
 	case strings.Contains(osImage, "gentoo"):
-		queryPackagesCmd = "qlist"
-		options = append(options, "-I")
-		packages = append(packages, "net-fs/nfs-utils", "sys-block/open-iscsi", "sys-fs/cryptsetup", "sys-fs/lvm2")
+		packageProbeExecutables["net-fs/nfs-utils"] = "mount.nfs"
+		packageProbeExecutables["sys-block/open-iscsi"] = "iscsiadm"
+		packageProbeExecutables["sys-fs/cryptsetup"] = "cryptsetup"
+		packageProbeExecutables["sys-fs/lvm2"] = "dmsetup"
 	default:
 		collectedData.conditions = types.SetCondition(collectedData.conditions, longhorn.NodeConditionTypeRequiredPackages, longhorn.ConditionStatusFalse,
 			string(longhorn.NodeConditionReasonUnknownOS),
 			fmt.Sprintf("Unable to verify the required packages because the OS image '%v' is unknown to the Longhorn system. Please ensure the required packages are installed.", osImage))
 		return
```

Comment:
I'm wondering if we can remove the message if Longhorn doesn't support the OS.

Reply:
WDYT? @COLDTURNIP @mantissahz

Reply:
Do we have another place/information to show users that the OS is unsupported except for official documents?
If users are aware of the unsupported OS and responsible for it when installing Longhorn, this message is annoying.

In general, I'm fine to remove it.

Reply:
> Do we have another place/information to show users that the OS is unsupported except for official documents?
https://longhorn.io/docs/1.8.0/best-practices/#operating-system

Yes, we don't need to handle the OSes that Longhorn doesn't support or handle.

Reply:
Agree to remove. Maybe we can show such warning on the UI, or we already have CLI preflight checker at least. Node condition is too deep for user to get notice.

Reply:
> Node condition is too deep for user to get notice.

Node conditions are good for troubleshooting and self-diagnosis. However, there are too many OSes in the market, so we only need to take care of the OSes mentioned in the official doc.

Reply:
Will append a commit to remove this condition for unsupported OSes.

Reply:
Isn't it removed?

Reply:
Yeah, we're going to remove the condition completely on unsupported OSes.